Engine/DataFeeds/LiveTradingDataFeed.cs

```diff
@@ -700,7 +627,61 @@ private IEnumerator<BaseData> GetNextTicksEnumerator()
         /// <filterpriority>1</filterpriority>
         public IEnumerator<TimeSlice> GetEnumerator()
         {
-            return _bridge.GetConsumingEnumerable(_cancellationTokenSource.Token).GetEnumerator();
+            var shouldSendExtraEmptyPacket = false;
+            // we want to emit to the bridge minimally once a second since the data feed is
+            // the heartbeat of the application, so this value will contain a second after
+            // the last emit time, and if we pass this time, we'll emit even with no data
+            var nextEmit = DateTime.MinValue;
+            var syncer = new SubscriptionSynchronizer(_universeSelection, _algorithm.TimeZone, _algorithm.Portfolio.CashBook, _frontierTimeProvider);
+            syncer.SubscriptionFinished += (sender, subscription) =>
+            {
+                RemoveSubscription(subscription.Configuration);
+                Log.Debug($"LiveTradingDataFeed.SubscriptionFinished(): Finished subscription: {subscription.Configuration} at {_algorithm.UtcTime} UTC");
+            };
+            while (!_cancellationTokenSource.IsCancellationRequested)
+            {
+                // perform sleeps to wake up on the second?
+                _frontierUtc = _timeProvider.GetUtcNow();
+                _frontierTimeProvider.SetCurrentTime(_frontierUtc);
+                TimeSlice timeSlice;
+                try
+                {
+                    timeSlice = syncer.Sync(Subscriptions);
+                }
+                catch (Exception err)
+                {
+                    Log.Error(err);
+                    // notify the algorithm about the error, so it can be reported to the user
+                    _algorithm.RunTimeError = err;
+                    _algorithm.Status = AlgorithmStatus.RuntimeError;
+                    shouldSendExtraEmptyPacket = true;
+                    break;
+                }
+                // check for cancellation
+                if (_cancellationTokenSource.IsCancellationRequested) break;
+                // emit on data or if we've elapsed a full second since last emit or there are security changes
+                if (timeSlice.SecurityChanges != SecurityChanges.None || timeSlice.Data.Count != 0 || _frontierUtc >= nextEmit)
+                {
+                    yield return timeSlice;
+                    // force emitting every second
+                    nextEmit = _frontierUtc.RoundDown(Time.OneSecond).Add(Time.OneSecond);
```

Comment:
no need to change -- simply noting that this appears to be an _old_ bug (probably has my name all over it :D )

consider `_frontierUtc` is one tick below the round second, `nextEmit` will end up being only one tick ahead of `_frontierUtc` -- now, this is unlikely to happen in non-tick algorithms since data is always emitted on rounded seconds, but for algorithms with many tick subscriptions this could become a source of undue spinning